components/library/RichTextEditor/Preview/RichTextEditor.tsx

```diff
@@ -0,0 +1,135 @@
+import { Box, Icon, InlineGrid, Text } from "@shopify/polaris";
+import { AlertCircleIcon } from "@shopify/polaris-icons";
+import { useEffect, useState } from "react";
+import type { ReactQuillProps } from "react-quill";
+import "react-quill/dist/quill.snow.css";
+import { ClientOnly } from "remix-utils/client-only";
+
+type ReducedQuillProps = Omit<ReactQuillProps, 'readOnly' | 'className' | 'theme' | 'children' | 'style' | 'tabIndex' | 'scrollingContainer'>
+
+const defaultModuleOptions = {
+    toolbar: [
+        ["bold", "italic", "underline", "blockquote"],
+        [
+            { list: "ordered" },
+            { list: "bullet" },
+            { indent: "-1" },
+            { indent: "+1" },
+        ],
+        ["link"],
+        ["clean"],
+    ],
+    clipboard: {
+        matchVisual: false // Needed to avoid this glitch: https://github.com/slab/quill/issues/2905
+    }
+};
+
+export function RichTextEditor({
+    bounds,
+    defaultValue,
+    formats,
+    id,
+    modules,
+    onBlur,
+    onChange,
+    onChangeSelection,
+    onFocus,
+    onKeyDown,
+    onKeyPress,
+    onKeyUp,
+    placeholder,
+    preserveWhitespace,
+    value,
+    label,
+    labelHidden = false,
+    disabled,
+    error,
+    fallbackComponent = <FallbackComponent />
+}: ReducedQuillProps & {
+    label: string;
+    labelHidden?: boolean;
+    disabled?: boolean;
+    error?: string | boolean | undefined;
+    fallbackComponent?: React.ReactNode
+}) {
+
+    const [Quill, setQuill] = useState<any>(null);
+
+    const mergedModuleOptions = {
+        ...defaultModuleOptions,
+        ...modules,
+        toolbar: modules?.toolbar || defaultModuleOptions.toolbar,
+        clipboard: {
+            ...defaultModuleOptions.clipboard,
+            ...(modules?.clipboard || {})
+        }
+    };
+
+
+    useEffect(() => {
+        if (typeof window !== "undefined") {
+            import("react-quill")
+                .then((QuillModule) => {
+                    if (QuillModule && QuillModule.default) {
+                        setQuill(() => QuillModule.default);
+                    } else {
+                        console.error("React Quill module or default export not found");
+                    }
+                })
+                .catch((e: any) => console.error("Failed to load react-quill", e));
+        }
+    }, []);
+
+    let className = disabled ? "quill--disabled " : "";
+    if (error) className += "quill--error";
+
+    return (
+        <ClientOnly fallback={fallbackComponent}>
```

Comment:
Let's remove this `ClientOnly` as well as the `remix-utils` package from the project since those are remix dependent and these should be able to be copy/pasted anywhere. With the `StatGrid` component you also need to use `ClientOnly` for it to work in Remix, but that's something external to the component that should be handled by the user. You can add some context in a comment for Remix-specific instructions

Reply:
See other comment.